tests/e2e/lib/k8s_common_helpers.go

```diff
@@ -0,0 +1,279 @@
+// output and error output, logging them if available. In case of errors, it returns an error message
+// with details about the issue.
+func ExecuteCommandInPodsSh(params ProxyPodParameters, command string) (string, string, error) {
+
+	var containerName string
+
+	kubeClient := params.KubeClient
+	kubeConfig := params.KubeConfig
+
+	if command == "" {
+		return "", "", fmt.Errorf("No command specified")
+	}
+
+	if kubeClient == nil {
+		return "", "", fmt.Errorf("No valid kubernetes.Clientset provided")
+	}
+
+	if kubeConfig == nil {
+		return "", "", fmt.Errorf("No valid rest.Config provided")
+	}
+
+	if params.PodName == "" {
+		return "", "", fmt.Errorf("No proxy pod specified for the command: %s", command)
+	}
+
+	if params.Namespace == "" {
+		return "", "", fmt.Errorf("No proxy pod namespace specified for the command: %s", command)
+	}
+
+	if params.ContainerName != "" {
+		containerName = params.ContainerName
+	} else {
+		containerName = "curl-tool"
+	}
+
+	log.Printf("Provided command: %s\n", command)
+	log.Printf("Command will run in the pod: %s, container: %s in the namespace: %s\n", params.PodName, containerName, params.Namespace)
+
+	option := &corev1.PodExecOptions{
+		Command:   strings.Split(command, " "),
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       true,
+		Container: containerName,
+	}
+
+	postRequest := kubeClient.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name(params.PodName).
+		Namespace(params.Namespace).
+		SubResource("exec").
+		VersionedParams(option, scheme.ParameterCodec)
+
+	log.Printf("Full command URL: %s\n", postRequest.URL())
+
+	executor, err := remotecommand.NewSPDYExecutor(kubeConfig, "POST", postRequest.URL())
+	if err != nil {
+		return "", "", err
+	}
+
+	stdOutput := &bytes.Buffer{}
+	stdErrOutput := &bytes.Buffer{}
+
+	err = executor.Stream(remotecommand.StreamOptions{Stdout: stdOutput, Stderr: stdErrOutput})
+
+	if stdOutput.Len() > 0 {
+		log.Printf("stdOutput: %s\n", stdOutput.String())
+	}
+
+	if stdErrOutput.Len() > 0 {
+		log.Printf("stdErrOutput: %s\n", stdErrOutput.String())
+	}
+
+	if err != nil {
+		log.Printf("Error while streaming command output: %v\n", err)
+		return stdOutput.String(), stdErrOutput.String(), fmt.Errorf("Error while streaming command output: %v", err)
+	}
+
+	return stdOutput.String(), stdErrOutput.String(), nil
+}
+
+// GetFirstPodByLabel retrieves a first found pod in the specified namespace based on the given label selector.
+// It uses the provided Kubernetes client to interact with the Kubernetes API.
+//
+// Parameters:
+//   - clientset: A pointer to the Kubernetes client (*kubernetes.Clientset).
+//   - namespace: The namespace in which to search for the pod.
+//   - labelSelector: The label selector to filter pods.
+//
+// Returns:
+//   - (*corev1.Pod, error): A pointer to the first pod matching the label selector, or an error if any.
+//
+// Example:
```

Comment:
I would remove examples from Doc comments. I think the code itself acts as example, but it harmless to let them

Reply:
Not everywhere, I would leave it for example here:

```
// ConvertJsonStringToURLParams takes a JSON string as input and converts it to URL-encoded parameters.
// It returns a string containing the URL-encoded parameters.
//
// Parameters:
//   - payload (string): The JSON string to be converted.
//
// Returns:
//   - string: The URL-encoded parameters.
//   - error: An error, if any, during the conversion process.
//
// Example:
//
//	input:  `{"name": "John", "age": 30, "city": "New York"}`
//	output: `&{name=John&age=30&city=New+York}`

```

Reply:
Removed from the examples how to use functions, but left the above one, because it's important to know what example input/output is expected as payload.